src/huggingface_hub/hf_api.py

```diff
@@ -8261,6 +8304,361 @@ def grant_access(
+            url=webhook_data["url"],
+            secret=webhook_data.get("secret"),
+            domains=webhook_data["domains"],
+            disabled=webhook_data["disabled"],
+        )
+
+        return webhook
+
+    @validate_hf_hub_args
+    def list_webhooks(self, token: Optional[str] = None) -> List[Webhook]:
+        """List all configured webhooks.
+
+        Args:
+            token (`str`, *optional*):
+                A valid authentication token (see https://huggingface.co/settings/token).
+
+        Returns:
+            `List[Webhook]`:
+                A list of [`huggingface_hub.hf_api.Webhook`] configurations.
+
+        Example:
+            ```python
+            >>> webhooks = list_webhooks()
+            >>> len(webhooks)
+            2
+            >>> webhooks[0]
+            Webhook(
+                id="654bbbc16f2ec14d77f109cc",
+                watched=[WebhookWatchedItem(type="user", name="julien-c"), WebhookWatchedItem(type="org", name="HuggingFaceH4")],
+                url="https://webhook.site/a2176e82-5720-43ee-9e06-f91cb4c91548",
+                secret="my-secret",
+                domains=["repo", "discussion"],
+                disabled=False,
+            )
+            ```
+        """
+        response = get_session().get(
+            f"{ENDPOINT}/api/settings/webhooks",
+            headers=self._build_hf_headers(token=token),
+        )
+        hf_raise_for_status(response)
+        webhooks_data = response.json()
+
+        return [
+            Webhook(
+                id=webhook["id"],
+                watched=[WebhookWatchedItem(type=item["type"], name=item["name"]) for item in webhook["watched"]],
+                url=webhook["url"],
+                secret=webhook.get("secret"),
+                domains=webhook["domains"],
+                disabled=webhook["disabled"],
+            )
+            for webhook in webhooks_data
+        ]
+
+    @validate_hf_hub_args
+    def create_webhook(
+        self,
+        watched: List[WebhookWatchedItem],
+        url: str,
+        domains: List[WEBHOOK_DOMAIN_T],
+        secret: Optional[str] = None,
+        token: Optional[str] = None,
+    ) -> Webhook:
+        """Create a new webhook.
+
+        Args:
+            watched (`List[WebhookWatchedItem]`):
+                List of [`WebhookWatchedItem`] to be watched by the webhook. It an be users, orgs, models, datasets or spaces.
+            url (`str`):
+                URL to send the payload to.
+            domains (`List[Literal["repo", "discussion"]]`):
+                List of domains to watch. It can be "repo", "discussion" or both.
+            secret (`str`, *optional*):
+                A secret to sign the payload with.
+            token (`str`, *optional*):
+                A valid authentication token (see https://huggingface.co/settings/token).
+
+        Returns:
+            [`Webhook`]:
+                The created [`huggingface_hub.hf_api.Webhook`] configuration.
```

Comment:
Probably i am overdoing it with references :smile:

I was confused with the referencing style: Is it always enough to reference just like [`WebhookInfo`] then? If yes, why are there examples of [`huggingface_hub.hf_api.RepoSibling`] or [`hf_api.DatasetInfo`]? Is there a need of standardising this as well? Things like this increases in my opinion fraction for contributors to easily decide which way to go (only if they do not know it anyway of course :D).

Reply:
What I meant is that in
```py
        Returns:
            [`Webhook`]:
                The created [`huggingface_hub.hf_api.Webhook`] configuration.
```

there is no need to reference twice the `Webhook` object (though it's valid).
Regarding the format,  
```
[`Webhook`]
```
and
```
[`huggingface_hub.hf_api.Webhook`]
```
are both valid as the doc builder will be able to find them in both cases. I usually prefer the short version for readability. Something you have to use the long version because the object is not exposed at the root of the package. Tough most of the time, it is more for legacy reasons than a proper reason.